front/src/reducers/osrdconf/osrdConfCommon/index.ts

```diff
@@ -176,25 +177,40 @@ export function buildCommonConfReducers<S extends OsrdConfState>(): CommonConfRe
       const feature = omit(action.payload.feature, ['_vectorTileFeature']);
       state.featureInfoClick = { ...action.payload, feature };
     },
-    updatePathSteps(
-      state: Draft<S>,
-      action: PayloadAction<{ pathSteps: S['pathSteps']; resetPowerRestrictions?: boolean }>
-    ) {
-      state.pathSteps = action.payload.pathSteps;
-      if (action.payload.resetPowerRestrictions) {
-        state.powerRestriction = [];
-      }
+    // update path steps without changing the itinerary (only add vias on the existing pathfinding,
```

Comment:
shouldn’t we use replaceItinerary if we add a vias (and change the number of elements) ?
I think the more important distinction is the number of elements.

some use cases only add a flag to the pathSteps or update the coordinates, but we keep exactly the same start, vias, end. in that case the itinerary remains exactly the same (same number of elements, all the elements are the same) 

if we add a vias on the existing itinerary, the train uses exactly the same path, but it might stop somewhere it didn’t before.

Reply:
Ok nice, that’ll be helpful for my fix on double calls on the pathfinding endpoint

Reply:
Ok nice, that’ll be helpful for my fix on double calls on the pathfinding endpoint

Reply:
can you comment both action to specify when they are used, and that one doesn’t require to recall pathfinding endpoint, and the other does ?